G-Research.OpenTelemetry.Processor.Partial.Tests/PartialActivityProcessorTests.cs

```diff
@@ -9,29 +10,56 @@ public class PartialActivityProcessorTests
     {
         private List<LogRecord> exportedLogs = [];
         private InMemoryExporter<LogRecord> logExporter;
-        private PartialActivityProcessor processor;
+        private PartialActivityProcessor _processor;
         private const int HeartbeatIntervalMilliseconds = 1000;
 
         public PartialActivityProcessorTests()
         {
             logExporter = new InMemoryExporter<LogRecord>(exportedLogs);
-            processor = new PartialActivityProcessor(logExporter, HeartbeatIntervalMilliseconds);
+            _processor = new PartialActivityProcessor(logExporter, HeartbeatIntervalMilliseconds);
+        }
+
+        [Fact]
+        public void Log_ShouldContainDefinedResource()
```

Comment:
This test does not seem to test the logic for adding the resource. I think we should instead test with a Provider?

Reply:
this is testing whether exported log has resource that was added. i'll try with provider instead.

Reply:
i am not sure how to test this, because it seems that otlp exporter adds resource to the end result. any ideas?

Reply:
i think i found a way